File: app/src/main/java/com/rakangsoftware/recyclerviewdemo/CountryRepository.java
package com.rakangsoftware.recyclerviewdemo;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

public class CountryRepository {

    public List<Country> getAll() {
        List<Country> countries = new ArrayList<>();

        Locale[] locales = Locale.getAvailableLocales();
        for (Locale locale : locales) {
            String countryCode = locale.getCountry();
            String countryName = locale.getDisplayCountry();
            if (!countryCode.trim().isEmpty() && !countryName.trim().isEmpty()) {
                Country country = new Country(
                        countryCode,
                        countryName
                );
                countries.add(country);
            }
        }

        return countries;
    }

}

File: app/src/main/java/com/rakangsoftware/recyclerviewdemo/CountryRepository.kt
package com.rakangsoftware.recyclerviewdemo

import java.util.ArrayList
import java.util.Locale

class CountryRepository {

    val all: List<Country>
        get() {
            val countries = ArrayList<Country>()
            val locales = Locale.getAvailableLocales()
            for (locale in locales) {
                val countryCode = locale.country
                val countryName = locale.displayCountry
                if (!countryCode.trim { it <= ' ' }.isEmpty() && !countryName.trim { it <= ' ' }.isEmpty()) {
                    val country = Country(
                            countryCode,
                            countryName
                    )
                    countries.add(country)
                }
            }

            return countries
        }

}
